velox/common/encode/Base64.h

```diff
@@ -32,125 +29,119 @@ class Base64 {
   static const size_t kCharsetSize = 64;
   static const size_t kReverseIndexSize = 256;
 
-  /// Character set used for encoding purposes.
+  /// Character set used for Base64 encoding.
   /// Contains specific characters that form the encoding scheme.
   using Charset = std::array<char, kCharsetSize>;
 
-  /// Reverse lookup table for decoding purposes.
+  /// Reverse lookup table for decoding.
   /// Maps each possible encoded character to its corresponding numeric value
   /// within the encoding base.
   using ReverseIndex = std::array<uint8_t, kReverseIndexSize>;
 
-  /// Padding character used in encoding.
-  static const char kPadding = '=';
-
-  /// Encodes the specified number of characters from the 'data'.
-  static std::string encode(const char* data, size_t len);
-
+  /// Encodes the specified number of characters from the 'input'.
+  static std::string encode(const char* input, size_t inputSize);
   /// Encodes the specified text.
   static std::string encode(folly::StringPiece text);
-
   /// Encodes the specified IOBuf data.
-  static std::string encode(const folly::IOBuf* text);
-
-  /// Returns encoded size for the input of the specified size.
-  static size_t calculateEncodedSize(size_t size, bool withPadding = true);
-
-  /// Encodes the specified number of characters from the 'data' and writes the
-  /// result to the 'output'. The output must have enough space, e.g. as
+  static std::string encode(const folly::IOBuf* inputBuffer);
+  /// Encodes the specified number of characters from the 'input' and writes the
+  /// result to the 'outputBuffer'. The output must have enough space as
   /// returned by the calculateEncodedSize().
-  static void encode(const char* data, size_t size, char* output);
-
-  /// Decodes the specified encoded text.
-  static std::string decode(folly::StringPiece encoded);
+  static void encode(const char* input, size_t inputSize, char* outputBuffer);
 
-  /// Returns the actual size of the decoded data. Will also remove the padding
-  /// length from the input data 'size'.
-  static size_t calculateDecodedSize(const char* data, size_t& size);
-
-  /// Decodes the specified number of characters from the 'data' and writes the
-  /// result to the 'output'. The output must have enough space, e.g. as
-  /// returned by the calculateDecodedSize().
-  static void decode(const char* data, size_t size, char* output);
+  /// Encodes the input data using Base64 URL encoding.
+  static std::string encodeUrl(const char* input, size_t inputSize);
+  static std::string encodeUrl(folly::StringPiece text);
+  static std::string encodeUrl(const folly::IOBuf* inputBuffer);
```

Comment:
Please retain the comments from before as they clarify the semantics.

Reply:
Reverted the changes.